### DISCUSSION TITLE
Poking the bear

### DISCUSSION THREAD

**Discussion by doobes** [TriliumNext/Notes#166] [General]
Greetings all,

Please excuse my intrusion here.

I'm a complete noob to Trilium, yet I'm also completely smitten by it and the promise it holds.

So much so, that I've volunteered to be on the dev team, if only in a support role as I'm definitely not a dev. My expertise is in program/project development and perhaps this is where I can be the most help. I'm an electro-mechanical engineer and have been building things (mostly underwater robotic systems) for over 40 years.

I've read through the discussions, and while there seemed to be a great ground swell of enthusiasm earlier in the year, discussions lately don't seem to reflect that level of excitement.  I get it, life happens, etc, etc, etc.

So, I have some questions/comments:

I've NEVER been involved in something (a collaborative software development process) like this, how does it work? 

Is there a governing committee (or person) who charts the path forward? If there isn't shouldn't there be?

Can we please start a dialog if for no other reason than to reignite the enthusiasm expressed earlier in the year as to how to move forward? If so, what's the most expeditious means of doing that?

I've got a number of thoughts, but they are definitely those of an outside looking in frame of mind and may be completely harebrained, but who knows? In my years of building things, oftentimes the best ideas come from those with no intrinsic knowledge of the task at hand.

I am retired and have time to allocate to the project. 

Take care

chris

**Comment by meichthys** [comment]
@doobes I'm with you on this. In general, there's no 'one right way' for the development process, and we don't really have organization other than a Github organization which eliandoran started. We are basically a blank slate - we do not have a governing committee (although that may not be a bad idea). Currently the only structure we have to what we call TriliumNext is a group of people that are interested in the project. Some of the people have different permissions in the TriliumNext github organization which defines what specific users can do inside the github organization. You can see the membership/ownership [here](https://github.com/orgs/TriliumNext/people?query=.

There was a github discussion about where to discuss, but that didn't materialize so we currently have two discussions happening on [Discord](https://discord.gg/V56E6umeeE) and [Matrix](https://app.element.io/#/room/#trilium:matrix.org).

If you have the time and enthusiasm to push us forward, please, by all means, do it!

**Reply by zerebos** [reply]
Here is an updated link for the Discord at least: https://discord.gg/V56E6umeeE

**Reply by doobes** [reply]
H'mmm, 

I obviously buggered something up:

![image](https://github.com/TriliumNext/Notes/assets/23409761/9a3d1802-7c08-4186-971a-e362a68958b6)

**Reply by zerebos** [reply]
Hmmm is your Discord account brand new? I've never seen that before outside of those that are banned. But this server has no bans yet....

**Reply by doobes** [reply]
Tried again on another machine and managed to get registered.  The daemon is more than a little flaky....

**Comment by CobriMediaJulien** [comment]
I agree, that a more structured organisation could help the former development. Why we dont build smaller teams and then the team can decide where to communite but we organise the rest via other tools. There are plenty of project managemnt tools to handle, rate and decide about feature request. If we have this collection teams can form by themselve and then organise themself. but a roadmap, list of teams and some project management could help?

Github -> Main form of communication in public. Projectmanagement tools for organising the goals. i think one of the biggest downsides right now is, that nobody has a overview. At least im in matrix chat and discord and lost it.

**Comment by greg-offerfit** [comment]
> I've NEVER been involved in something (a collaborative software development process) like this, how does it work?

The pessimistic answer is that this *is* how it works, most of the time. A bunch of people get together all excited about the new project, talk a lot, don't get much done, enthusiasm wanes, project goes nowhere. 

If something's going to happen, it's because someone is going to start actually doing something, but that's kinda the problem - there isn't really a clear place to start. The app itself is pretty solid. There's improvements, sure, but the core is there, and it works. zadam is still pushing out small iterations of fixes and patches (last release was 3weeks ago), so it's not even been formally abandoned. There's just not much to actually do with it.

**Reply by CobriMediaJulien** [reply]
A proper Plug In system would be the biggest benefit right now for more collaboration. But just my opinion.

**Comment by root-hal9000** [comment]
Just saw this discussion now, but in case you haven't seen, we are getting together to discuss some of these issues - See discussion #187 for an invite to the kick-off meeting

**Comment by meichthys** [comment]
Closing the discussion here since we have had our official first meeting, and most of the questions have been addressed since then or at the meeting [(meeting minutes)](https://github.com/TriliumNext/.github-private/blob/main/meeting_minutes/2024-07-07-minutes.md).
